src/gui/executionlogwidget.cpp

```diff
@@ -29,78 +29,73 @@
 #include "executionlogwidget.h"
 
 #include <QDateTime>
+#include <QMenu>
 #include <QPalette>
 
 #include "base/global.h"
-#include "loglistwidget.h"
+#include "base/logger.h"
+#include "log/logfiltermodel.h"
+#include "log/loglistview.h"
+#include "log/logmodel.h"
 #include "ui_executionlogwidget.h"
 #include "uithememanager.h"
 
-ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes &types)
+ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes types)
     : QWidget(parent)
     , m_ui(new Ui::ExecutionLogWidget)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES, types, this))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, Log::ALL, this))
+    , m_messageModel(new LogMessageModel(this))
+    , m_peerModel(new LogPeerModel(this))
+    , m_messageFilterModel(new LogFilterModel(types, this))
+    , m_messageView(new LogListView(this))
+    , m_peerView(new LogListView(this))
 {
     m_ui->setupUi(this);
 
+    m_messageFilterModel->setSourceModel(m_messageModel);
+    m_messageView->setModel(m_messageFilterModel);
+    m_messageView->setContextMenuPolicy(Qt::CustomContextMenu);
+    connect(m_messageView, &LogListView::customContextMenuRequested, this, [this](const QPoint &pos) {
+        displayContextMenu(pos, m_messageView, m_messageModel);
+    });
+
+    m_peerView->setModel(m_peerModel);
+    m_peerView->setContextMenuPolicy(Qt::CustomContextMenu);
+    connect(m_peerView, &LogListView::customContextMenuRequested, this, [this](const QPoint &pos) {
+        displayContextMenu(pos, m_peerView, m_peerModel);
+    });
+
+    m_ui->tabGeneral->layout()->addWidget(m_messageView);
+    m_ui->tabBan->layout()->addWidget(m_peerView);
+
 #ifndef Q_OS_MACOS
     m_ui->tabConsole->setTabIcon(0, UIThemeManager::instance()->getIcon("view-calendar-journal"));
     m_ui->tabConsole->setTabIcon(1, UIThemeManager::instance()->getIcon("view-filter"));
 #endif
-    m_ui->tabGeneral->layout()->addWidget(m_msgList);
-    m_ui->tabBan->layout()->addWidget(m_peerList);
-
-    const Logger *const logger = Logger::instance();
-    for (const Log::Msg &msg : asConst(logger->getMessages()))
-        addLogMessage(msg);
-    for (const Log::Peer &peer : asConst(logger->getPeers()))
-        addPeerMessage(peer);
-    connect(logger, &Logger::newLogMessage, this, &ExecutionLogWidget::addLogMessage);
-    connect(logger, &Logger::newLogPeer, this, &ExecutionLogWidget::addPeerMessage);
 }
 
 ExecutionLogWidget::~ExecutionLogWidget()
 {
     delete m_ui;
 }
 
-void ExecutionLogWidget::showMsgTypes(const Log::MsgTypes &types)
+void ExecutionLogWidget::setMsgTypes(const Log::MsgTypes types)
```

Comment:
Please use "message" instead of "msg" whenever possible.

Reply:
changed the member function of `ExecutionLogWidget`, do I need to change the member function of `MainWindow` as well (`MainWindow::setExecutionLogMsgTypes` and `MainWindow::executionLogMsgTypes` )

Reply:
I don't insist that you change anything except the ones you touch anyway.